openssl/src/asn1.rs

```diff
@@ -342,12 +342,16 @@ impl fmt::Display for Asn1ObjectRef {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         unsafe {
             let mut buf = [0; 80];
+            // NOTE: `len` may be greater than the size of the buffer we provided, which would
+            // indicate that the result is truncated. See docs here:
+            // https://www.openssl.org/docs/man1.0.2/man3/OBJ_obj2txt.html
```

Comment:
If this is true, then the slicing down on line 355 is going to panic.

Reply:
Fixed and also added handling for the error case.